tests/integration/_experimental/object_detection/test_evaluator_multiclass_test_suite_metrics.py

```diff
@@ -0,0 +1,320 @@
+from .test_evaluator_multiclass_test_sample_metrics import TEST_CONFIGURATIONS
+from .test_evaluator_multiclass_test_sample_metrics import TEST_DATA
+from kolena._experimental.object_detection import ObjectDetectionEvaluator
+from kolena._experimental.object_detection import TestCase
+from kolena._experimental.object_detection import TestSuite
+from kolena._experimental.object_detection.workflow import ClassMetricsPerTestCase
+from kolena._experimental.object_detection.workflow import TestCaseMetrics
+from kolena._experimental.object_detection.workflow import TestCaseMetricsSingleClass
+from kolena._experimental.object_detection.workflow import TestSuiteMetrics
+from tests.integration.helper import with_test_prefix
+
+
+TEST_CASE_1 = TestCase(with_test_prefix("test_evaluator_multiclass_suite_1"), reset=True)
+TEST_CASE_2 = TestCase(with_test_prefix("test_evaluator_multiclass_suite_2"), reset=True)
+
+
+@pytest.mark.metrics
+@pytest.mark.parametrize(
+    "config_name, test_case_metrics, expected",
+    [
+        (
+            "Threshold: Fixed(0.5) by class, IoU: 0.5, confidence ≥ 0.0",
+            [
+                (
+                    TEST_CASE_1,
+                    TestCaseMetrics(
+                        PerClass=[],
+                        Objects=0,
+                        Inferences=0,
+                        TP=0,
+                        FN=0,
+                        FP=0,
+                        macro_Precision=0,
+                        macro_Recall=0,
+                        macro_F1=0,
+                        macro_AP=0,
+                        mean_AP=0,
+                    ),
+                ),
+            ],
+            TestSuiteMetrics(
+                n_images=2,
+                mean_AP=0,
+            ),
+        ),
+        (
+            "Threshold: Fixed(0.5) by class, IoU: 0.5, confidence ≥ 0.0",
+            [
+                (
+                    TEST_CASE_1,
+                    TestCaseMetrics(
+                        PerClass=[
+                            ClassMetricsPerTestCase(
+                                Class="test",
+                                nImages=1,
+                                Threshold=1,
+                                Objects=1,
+                                Inferences=1,
+                                TP=1,
+                                FN=1,
+                                FP=1,
+                                Precision=1,
+                                Recall=1,
+                                F1=1,
+                                AP=1,
+                            ),
+                        ],
+                        Objects=0,
+                        Inferences=0,
+                        TP=0,
+                        FN=0,
+                        FP=0,
+                        macro_Precision=0,
+                        macro_Recall=0,
+                        macro_F1=0,
+                        mean_AP=0.123,
```

Comment:
Hmm this mAP value doesn't make sense to me. I thought `mean_AP` is calculated by taking average of `AP` across all labels, so in this case shouldn't it be 1? Or did I miss something?

Reply:
Outdated, but yes the `mean_AP` was a dummy number to just test the behaviour of the function - here it averages the APs across test cases.